Mathlib/Tactic/NormNum/RealSqrt.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2025 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+import Mathlib.Data.Real.Sqrt
+
+/-! # `norm_num` extension for `Real.sqrt`
+
+This module defines a `norm_num` extension for `Real.sqrt` and `NNReal.sqrt`.
+
+## Implementation notes
+
+While the extension for `Real.sqrt` can handle both integers and rationals, the one for
+`NNReal.sqrt` can only deal with integers, due to a limitation of norm_num
+(i.e. the `IsRat` type requires a Ring instance).
+-/
+
+namespace Tactic.NormNum
+
+open Qq Lean Lean.Meta Elab.Tactic Mathlib.Meta.NormNum NNReal
+
+lemma isNat_realSqrt {x : ℝ} {nx ny : ℕ} (h : IsNat x nx) (hy : ny * ny = nx) :
+    IsNat √x ny := ⟨by simp [h.out, ← hy]⟩
+
+lemma isNat_nnrealSqrt {x : ℝ≥0} {nx ny : ℕ} (h : IsNat x nx) (hy : ny * ny = nx) :
+    IsNat (NNReal.sqrt x) ny := ⟨by simp [h.out, ← hy]⟩
+
+lemma isNat_realSqrt_neg {x : ℝ} {nx : ℕ} (h : IsInt x (Int.negOfNat nx)) :
+    IsNat √x 0 := by
+  refine ⟨?_⟩
+  simp only [Nat.cast_zero]
+  refine Real.sqrt_eq_zero_of_nonpos ?_
+  simp [h.out]
+
+lemma isNat_realSqrt_of_isRat_negOfNat {x : ℝ} {num : ℕ} {denom : ℕ}
+    (h : IsRat x (.negOfNat num) denom) : IsNat √x 0 := by
+  refine ⟨?_⟩
+  obtain ⟨inv, rfl⟩ := h
+  simp only [Nat.cast_zero]
+  refine Real.sqrt_eq_zero_of_nonpos ?_
+  simp only [Int.cast_negOfNat, neg_mul, neg_nonpos]
+  exact mul_nonneg (Nat.cast_nonneg' _) (invOf_nonneg.2 <| Nat.cast_nonneg' _)
```

Comment:
```suggestion
refine ⟨?_⟩
  obtain ⟨inv, rfl⟩ := h
  simpa only [Nat.cast_zero, Real.sqrt_eq_zero', Int.cast_negOfNat, neg_mul, neg_nonpos] using
    mul_nonneg (Nat.cast_nonneg' _) (invOf_nonneg.2 <| Nat.cast_nonneg' _)
```

Reply:
I think it would be better to insert a `have` and let `positivity` solve the last goal.

Reply:
positivity doesn't solve that goal, at least it's not immediate.